.github/workflows/build.yml

```diff
@@ -0,0 +1,68 @@
+name: Build to GitHub pages
+on:
+  push:
+    branches:
+      - master 
+  pull_request:
+    branches:
+      - master 
+
+jobs: 
+  build:
+    runs-on: ubuntu-18.04
+    steps:
+      ### PREP ###
+      - name: Checkout repo
+        uses: actions/checkout@v2.3.1
+        with:
+          persist-credentials: false # needed for JamesIves/github-pages-deploy-action
+
+      ### RUBY ###
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6.3
+      - name: Cache Ruby gems
+        id: cache-gems
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-1-${{ hashFiles('Gemfile.lock') }}
+      - name: Set Ruby gems directory
+        run: bundle config path vendor/bundle
+      - name: Install Ruby gems
+        run: bundle install --jobs 4 --retry 3 --deployment
+      - name: Install node 
+        run: npm install 
+
+      ### BUILD ###
+      - name: Build Jekyll 
+        run: bundle exec rake build 
+
+      ### TEST ###
+      - name: Cache htmltest files
+        uses: actions/cache@v2
+        with:
+          path: tmp/htmltest
+          key: htmltest-1
+      - name: Test built site with htmltest
+        uses: wjdp/htmltest-action@v0.13.0-rc1 
+        continue-on-error: true
```

Comment:
This will ignore any failure of htmltest, you probably want to remove. Had this set on history project due to some issue and need to remove it there.